File: app/src/main/java/com/example/onboardingscreen/ViewPagerAdapter.java
package com.example.onboardingscreen;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import com.airbnb.lottie.LottieAnimationView;

public class ViewPagerAdapter extends PagerAdapter {
    Context context;
    int sliderAllAnimation[] = {R.raw.one, R.raw.two, R.raw.three};
    int sliderAllTitle[] = {R.string.screen1, R.string.screen2, R.string.screen3};
    int sliderAllDesc[] = {R.string.screen1desc, R.string.screen2desc, R.string.screen3desc};

    public ViewPagerAdapter(Context context) {
        this.context = context;
    }

    @Override
    public int getCount() {
        return sliderAllTitle.length;
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == (LinearLayout) object;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {

        LayoutInflater layoutInflater = (LayoutInflater) context.getSystemService(context.LAYOUT_INFLATER_SERVICE);
        View view = layoutInflater.inflate(R.layout.slider_screen, container, false);

        LottieAnimationView sliderAnimation = (LottieAnimationView) view.findViewById(R.id.sliderAnimation);
        TextView sliderTitle = (TextView) view.findViewById(R.id.sliderTitle);
        TextView sliderDesc = (TextView) view.findViewById(R.id.sliderDesc);

        sliderAnimation.setAnimation(sliderAllAnimation[position]);
        sliderTitle.setText(this.sliderAllTitle[position]);
        sliderDesc.setText(this.sliderAllDesc[position]);

        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((LinearLayout) object);
    }
}

File: onboardingscreenkotlin/src/main/java/com/example/onboardingscreenkotlin/ViewPagerAdapter.kt
package com.example.onboardingscreenkotlin

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.viewpager.widget.PagerAdapter
import com.airbnb.lottie.LottieAnimationView

class ViewPagerAdapter(var context: Context) : PagerAdapter() {
    var sliderAllAnimation = intArrayOf(R.raw.one, R.raw.two, R.raw.three)
    var sliderAllTitle = intArrayOf(R.string.screen1, R.string.screen2, R.string.screen3)
    var sliderAllDesc = intArrayOf(R.string.screen1desc, R.string.screen2desc, R.string.screen3desc)
    override fun getCount(): Int {
        return sliderAllTitle.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object` as LinearLayout
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val layoutInflater =
            context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = layoutInflater.inflate(R.layout.slider_screen, container, false)
        val sliderAnimation = view.findViewById<View>(R.id.sliderAnimation) as LottieAnimationView
        val sliderTitle = view.findViewById<View>(R.id.sliderTitle) as TextView
        val sliderDesc = view.findViewById<View>(R.id.sliderDesc) as TextView
        sliderAnimation.setAnimation(sliderAllAnimation[position])
        sliderTitle.setText(sliderAllTitle[position])
        sliderDesc.setText(sliderAllDesc[position])
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as LinearLayout)
    }
}